test/test.globals.js

```diff
@@ -284,3 +287,45 @@ lab.test('varson setting', async() => {
   expect(context).to.equal({ yaml1: { test1: true } });
   await server.stop();
 });
+
+lab.test('?json=1', async () => {
+  const server = new Hapi.Server({
+    debug: { request: '*', log: 'hapi-views' }
+  });
+  server.method('yaml', (request, yamlFile) => {
+    return new Promise((resolve) => {
+      return resolve({ test1: true });
+    });
+  });
+  await server.register([
+    require('vision'),
+    {
+      plugin: require('../'),
+      options: {
+        debug: true,
+        dataPath: `${process.cwd()}/test/yaml`,
+        varsonSettings: {
+          start: '{{',
+          end: '}}'
+        },
+        routes: {
+          '/yaml': {
+            view: 'yaml',
+            data: {
+              yaml1: '{{methods.yaml()}}',
+            }
+          }
+        }
+      }
+    }
+  ]);
+  server.views({
+    engines: { html: require('handlebars') },
+    path: `${__dirname}/views`
+  });
+  await server.start();
+  // tests
+  const response = await server.inject({ url: '/yaml?json=1' });
+  expect(response.headers['content-type']).to.contain('application/json');
```

Comment:
This test isn't accurate. Needs to test output

Reply:
Git doesn't show it here, but I added this: https://github.com/firstandthird/hapi-views/blob/ab8d9e4e1008f7ba60d7c5a408c958ec4e00fe65/test/test.globals.js#L330